design/one-pager-package-format-v2.md

```diff
@@ -0,0 +1,267 @@
+consumers - a tool to build platforms. Crossplane enables a platform builder to
+[compose and offer][composition] their own APIs by defining composite resources.
+A composite resource is a Kubernetes resource that is reconciled by rendering
+other resources. A composite resource may consist of other composite resources,
+but the 'leaves' of these compositions are always 'managed resources' - a form
+of Kubernetes resource that is reconciled by a Crossplane provider.
+
+Several conceptual and API changes are required to focus Crossplane on providing
+a great experience around building, operating, and consuming platforms. Part of
+this simplification involves focusing packages and the package manager on two
+key use cases; extending Crossplane with new providers, and extending Crossplane
+with support for new composite resources.
+
+## Goals
+
+The goal of this document is to supplement the existing [package manager v2
+design][packages-v2] by formalising the content of a package - the filesystem
+that exists inside a package's OCI image. This metadata provides two purposes:
+
+1. It provides the information the Crossplane package manager needs to manage
+   the lifecycle of a package - to install or upgrade a provider or a
+   configuration.
+1. It describes what is packaged. This information is not used by Crossplane,
+   but provides an API that other tools such as package registries can use to
+   introspect a package.
+
+Note that one goal of this design is to build a user experience around working
+with 'providers' and 'configurations', rather than working with 'packages'. The
+package itself will be positioned as strictly the delivery mechanism.
+
+## Proposal
+
+This document proposes that:
+
+1. All Crossplane packages contain a `crossplane.yaml` file at their root.
+1. The `crossplane.yaml` file describe either a `Provider` or a `Configuration`.
+1. Packages may include YAML representations of Crossplane specific custom
+   resources (e.g. a `Composition` or a managed resource CRD). Crossplane is not
+   opinionated about where in the package filesystem these files exist - the
+   package unpack process should detect them automatically.
+1. Additional metadata that is opaque to Crossplane be stored in a directory
+   named `.crossplane` at the root of the package, alongside `crossplane.yaml`.
+
+### Packaging a Provider
+
+A packaged Crossplane provider must be an OCI image consisting of:
+
+1. A `crossplane.yaml` file containing a `kind: Provider` at its root.
+1. One or more `CustomResourceDefinition` resources that must define Crossplane
+   provider specific resources, for example a managed resource.
+1. Optional additional metadata that is opaque to Crossplane in a `.crossplane`
+   directory at the root of the package.
+
+Below is an example `Provider`, at `v1alpha1`:
+
+```yaml
+# Required. Must be as below.
+apiVersion: meta.packages.crossplane.io/v1alpha1
```

Comment:
@negz my current proposal is with the new package types in `pkg.crossplane.io`, mainly to distinguish from existing `packages` and shorten the name a bit. If we go with that, how would you feel about:
```suggestion
apiVersion: meta.pkg.crossplane.io/v1alpha1
```

Reply:
@hasheddan Sounds good - I prefer `pkg` too, as long as we're consistent. I'll update this.